lib/os_places_api/client.rb

```diff
@@ -25,6 +25,7 @@ def locations_for_postcode(postcode)
         json = JSON.parse(response.body)
         raise UnexpectedResponse if json["results"].nil?
 
+        Postcode.create!(postcode: postcode, results: json["results"])
```

Comment:
There's a possibility (albeit unlikely) that two users could search for the same postcode at almost the same time.

If the second user's request is made after the first user's `find_by` executes but before the `create!`, we'll then error on the second user as the record already exists, but we've already passed the `find_by` logic at this point.

Would it be safer to use `create_or_update!` here instead?

Reply:
It would have to be simultaneous requests for an uncached postcode (of which there should be none, once the [cache is warmed up](https://trello.com/c/iy6P2rvX/2353-warm-up-locations-api-cache)), so I would be very surprised if this happened.

But one upside of the `create_or_update!` method is that it pre-emptively provides the code we'll need to [implement the cache warming up process](https://trello.com/c/PnzlTlbU/2352-implement-cache-warming-up-process-for-locations-api-5). I can see us adding some sort of `refresh_cache: true` parameter to the `locations_for_postcode` method, that forces an OS Places API request even when we have a record in the database, to skip the early return and force an update to the record.

Thanks, I'll make the change 👍

Reply:
Ah, did you mean `create_or_find_by!`?

```
     NoMethodError:
       undefined method `create_or_update!' for #<Class:0x00005581fb2e1e90>
       Did you mean?  create_or_find_by!
```

Reply:
It appears the [`create_or_update` method](https://apidock.com/rails/ActiveRecord/Base/create_or_update) I was thinking of had been deprecated long ago.

Reply:
I've pushed up a commit to fix the race condition 👍

Reply:
Nice, looks like a good fix.